spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -333,6 +352,14 @@ private Object doSendAndReceive(Object object, boolean shouldConvert) {
 					error = ((ErrorMessage) reply).getPayload();
 				}
 			}
+			if (reply == null && this.errorOnTimeout) {
```

Comment:
Hello @garyrussell 

This line has caused our code to break because we have a void method in our service activator so the value of ```reply``` is ```null``` and ```this.errorOnTimeout``` is rightly ```true```.

This is how we have defined our gateway:

```
    <int-http:inbound-gateway id="foosearch.service.rest.deleteFooGateWay"
                              request-channel="foosearch.service.rest.deleteFooChannel"
                              mapped-request-headers="*"
                              path="/v1/foosearch/delete"
                              request-payload-type="foo.platform.util.content.dto.IdentifierCollection"
                              message-converters="foosearch.service.rest.marshallingMessageConverter"
                              supported-methods="POST"/>

    <int:channel id="foosearch.service.rest.deleteFooChannel"/>

    <int:service-activator input-channel="foosearch.service.rest.deleteFooChannel"
                           ref="foosearch.service.core.foosearchservice" method="deleteFoo"/>
```

Is this a bug in Spring Integration or have we defined our Gateway incorrectly?

Does it have anything to do with https://jira.spring.io/browse/INT-3781?

Any advice you could offer would be most appreciated :smile: